clingwrapper/src/clingwrapper.cxx

```diff
@@ -1511,11 +1513,16 @@ Cppyy::TCppMethod_t Cppyy::GetMethodTemplate(
     Cppyy::AppendTypesSlow(proto, arg_types);
     Cppyy::AppendTypesSlow(explicit_params, templ_params);
 
-    Cppyy::TCppMethod_t cppmeth = Cpp::BestTemplateFunctionMatch(unresolved_candidate_methods, templ_params, arg_types);
-    
-    if(!cppmeth){
-        return nullptr;
-    }
+    Cppyy::TCppMethod_t cppmeth = nullptr;
+
+    if (unresolved_candidate_methods.size() == 1 && !templ_params.empty())
+      cppmeth =
+          Cpp::InstantiateTemplate(unresolved_candidate_methods[0],
```

Comment:
Why we try to instantiate the candidates. The overload resolution logic should do that automatically.

Reply:
We explicitly instantiate, only in one case. Where the potential candidate (i.e. overloads) functions is only one. And the user has specified the template argument explicitly with
```python
template_fn[arg1, ...](...)
```